roles/openshift_master/tasks/system_container.yml

```diff
@@ -17,20 +17,62 @@
   changed_when: "'Extracting' in update_result.stdout"
   when:
     - ("master" in result.stdout)
-    - (openshift.common.version is defined) and (openshift.common.version == openshift_version) | bool
+    - l_is_same_version
+    - not l_is_ha
```

Comment:
I'd suggest using blocks for the following conditions if possible:
- `not l_is_ha`
- `l_is_ha`
- `'master' in result.stdout`
- `not l_is_same version`

I would also suggest using `check_master_system_container` rather than `result` for the registered variable to avoid confusion.